File: 2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxK = function(nums) {
    let res = -1
    //creating a set
    let numsSet = new Set(nums);

    for (let num of numsSet) {
        if (numsSet.has(-num)) {
            res = Math.max(res, num);
        }
    }
    return res;
};

File: 2524-largest-positive-integer-that-exists-with-its-negative/largest-positive-integer-that-exists-with-its-negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        unordered_set<int> st(nums.begin(), nums.end());
        int maxNum = -1;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > maxNum && st.find(-nums[i]) != st.end())
                maxNum = nums[i];
        }

        return maxNum;
    }
};